PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:11 + add:0, change matches:
-    _inherit = ['ir.mail_server']
-    _inherit = ['ir.model']
-    _inherit = ["link.tracker"]
-    _inherit = ["link.tracker.click"]
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']
-    _inherit = ['res.users']
-    _inherit = ['utm.campaign']
-    _inherit = ['utm.medium']
-    _inherit = ['utm.source']

Total Changes: 26

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mass_mailing/models/ir_http.py
+++ b/addons/mass_mailing/models/ir_http.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ["ir.http"]
+    _inherit = "ir.http"
 
     @classmethod
     def _get_translation_frontend_modules_name(cls):

--- a/addons/mass_mailing/models/ir_mail_server.py
+++ b/addons/mass_mailing/models/ir_mail_server.py
@@ -6,7 +6,7 @@ from odoo.tools.misc import format_date
 
 
 class IrMail_Server(models.Model):
-    _inherit = ['ir.mail_server']
+    _inherit = 'ir.mail_server'
 
     active_mailing_ids = fields.One2many(
         comodel_name='mailing.mailing',

--- a/addons/mass_mailing/models/ir_model.py
+++ b/addons/mass_mailing/models/ir_model.py
@@ -5,7 +5,7 @@ from odoo import fields, models, _
 
 
 class IrModel(models.Model):
-    _inherit = ['ir.model']
+    _inherit = 'ir.model'
 
     is_mailing_enabled = fields.Boolean(
         string="Mailing Enabled",

--- a/addons/mass_mailing/models/link_tracker.py
+++ b/addons/mass_mailing/models/link_tracker.py
@@ -5,13 +5,13 @@ from odoo import api, fields, models
 
 
 class LinkTracker(models.Model):
-    _inherit = ["link.tracker"]
+    _inherit = "link.tracker"
 
     mass_mailing_id = fields.Many2one('mailing.mailing', string='Mass Mailing')
 
 
 class LinkTrackerClick(models.Model):
-    _inherit = ["link.tracker.click"]
+    _inherit = "link.tracker.click"
 
     mailing_trace_id = fields.Many2one('mailing.trace', string='Mail Statistics', index='btree_not_null')
     mass_mailing_id = fields.Many2one('mailing.mailing', string='Mass Mailing', index='btree_not_null')

--- a/addons/mass_mailing/models/mail_render_mixin.py
+++ b/addons/mass_mailing/models/mail_render_mixin.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class MailRenderMixin(models.AbstractModel):
-    _inherit = ["mail.render.mixin"]
+    _inherit = "mail.render.mixin"
 
     @api.model
     def _render_template_postprocess(self, model, rendered):

--- a/addons/mass_mailing/models/res_company.py
+++ b/addons/mass_mailing/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     def _get_social_media_links(self):
         self.ensure_one()

--- a/addons/mass_mailing/models/res_config_settings.py
+++ b/addons/mass_mailing/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     group_mass_mailing_campaign = fields.Boolean(
         string="Mailing Campaigns",

--- a/addons/mass_mailing/models/res_partner.py
+++ b/addons/mass_mailing/models/res_partner.py
@@ -5,5 +5,5 @@ from odoo import models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
     _mailing_enabled = True

--- a/addons/mass_mailing/models/res_users.py
+++ b/addons/mass_mailing/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import api, models, _
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     @api.model
     def _get_activity_groups(self):

--- a/addons/mass_mailing/models/utm_campaign.py
+++ b/addons/mass_mailing/models/utm_campaign.py
@@ -10,7 +10,7 @@ from odoo.tools.float_utils import float_round
 
 
 class UtmCampaign(models.Model):
-    _inherit = ['utm.campaign']
+    _inherit = 'utm.campaign'
 
     mailing_mail_ids = fields.One2many(
         'mailing.mailing', 'campaign_id',

--- a/addons/mass_mailing/models/utm_medium.py
+++ b/addons/mass_mailing/models/utm_medium.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmMedium(models.Model):
-    _inherit = ['utm.medium']
+    _inherit = 'utm.medium'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_mailings(self):

--- a/addons/mass_mailing/models/utm_source.py
+++ b/addons/mass_mailing/models/utm_source.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class UtmSource(models.Model):
-    _inherit = ['utm.source']
+    _inherit = 'utm.source'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_linked_mailings(self):
